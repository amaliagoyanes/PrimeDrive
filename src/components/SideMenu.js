import React, {Component} from 'react';
import {
    Text, TouchableOpacity, Dimensions, Picker, AsyncStorage, Alert
} from 'react-native';
import {Row, Col} from 'react-native-easy-grid';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import SwipeALot from 'react-native-swipe-a-lot';
import Modal from 'react-native-root-modal';
import {Actions} from 'react-native-router-flux';

import {IconButton, ButtonItem,} from './index';
import styles from '../style/styles';
import plainStyles from '../style/plain';
import {scale, scaleText} from '../scaling';


export default class SideMenu extends Component {
    constructor(props) {
        super(props);
        this.state = {
            open: false
        };

        this.close = this.close.bind(this);
        this.open = this.open.bind(this);
        this.openModal = this.openModal.bind(this);
    }

    open() {
        console.log('TYPE', this.props.printerType);
        this.setState({open: true})
    }

    close() {
        this.setState({open: false})
    }

    openModal(name) {
        this.setState({open: false});
        this.props.openModal(name);
    }

    render() {
        let header = {
            height: scale(40),
            marginTop: 30, paddingRight: 10, paddingLeft: 10,
            alignSelf: 'stretch',
        };
        return (
            <Modal
                onRequestClose={this.close}
                visible={this.state.open}
                style={styles.modal}
            >
                <TouchableOpacity onPress={this.close} style={[styles.modal, {
                    flexDirection: 'row',
                    justifyContent: 'flex-start',
                    flex: 1,
                }]}>
                    <TouchableOpacity activeOpacity={1} onPress={(event) => event.stopPropagation()} style={{
                        backgroundColor: '#404048',
                        width: Dimensions.get('screen').width / 3,
                        top: 0,
                        bottom: 0
                    }}>
                        <Row size={1} style={header}>
                            <Col size={2}>
                                <Text style={{color: "white", marginTop: 5}}>Menu</Text>
                            </Col>
                            <Col size={10}/>
                            <Col size={1}>
                                <TouchableOpacity onPress={this.close}
                                                  style={[styles.timesIcon, {
                                                      marginTop: 0,
                                                      backgroundColor: '#53535a'
                                                  }]}>
                                    <Icon size={scaleText(18)} name="close" color="white"/>
                                </TouchableOpacity>
                            </Col>
                            <Col size={0.25}/>
                        </Row>
                        <Row size={6.75}>
                            <Col style={[styles.slide, {marginTop: 0}]}>
                                <Row>
                                    <IconButton price="app.terminal" icon="printer" intl={this.props.intl}
                                                onPress={() => {
                                                    if (this.props.permissions && this.props.permissions['function_15']) {
                                                        this.openModal('terminalStatus');
                                                    } else {
                                                        this.props.alert(this.props.intl.messages["app.missingPermissions"]);
                                                    }
                                                }}/>
                                    <IconButton price="app.closedReceipts" icon="close-circle-outline"
                                                intl={this.props.intl}
                                                onPress={() => {
                                                    this.close();
                                                    this.props.openClosedReceipts();
                                                }}/>
                                    <IconButton price="app.parkedReceipts" icon="pause" intl={this.props.intl}
                                                onPress={() => {
                                                    this.close();
                                                    this.props.openParkedReceipts();
                                                }}/>
                                </Row>
                                <Row>
                                    <IconButton price="app.openDay" icon="calendar-clock" intl={this.props.intl}
                                                onPress={() => this.openModal('openDay')}/>
                                    <IconButton price="app.closeDay" icon="calendar-remove" intl={this.props.intl}
                                                onPress={() => {
                                                    if (this.props.cash_shift_id === null)
                                                        this.props.alert(this.props.intl.messages["app.noShiftWarning"]);
                                                    else if (this.props.permissions && this.props.permissions['function_13']) {
                                                        this.openModal('closeDay');
                                                    } else {
                                                        this.props.alert(this.props.intl.messages["app.missingPermissions"]);
                                                    }
                                                }}/>
                                    <IconButton price="app.checkInOut" icon="check-circle-outline"
                                                intl={this.props.intl}
                                                onPress={() => {
                                                    if (this.props.permissions && this.props.permissions['function_7']) {
                                                        this.openModal('checkInOut');
                                                    } else {
                                                        this.props.alert(this.props.intl.messages["app.missingPermissions"]);
                                                    }
                                                }}/>
                                </Row>
                                <Row>
                                    <IconButton price="app.printer" icon="printer" intl={this.props.intl}
                                                onPress={() => {
                                                    if (this.props.permissions && this.props.permissions['function_25']) {
                                                        if (!!this.props.printerType && this.props.printerType !== 'mpop')
                                                            this.openModal('netprinterStatus');
                                                        else
                                                            this.openModal('printerStatus');
                                                    } else {
                                                        this.props.alert(this.props.intl.messages["app.missingPermissions"]);
                                                    }
                                                }}/>
                                    {/*<IconButton price="app.productionPrinter" icon="printer" intl={this.props.intl}*/}
                                                {/*onPress={() => this.openModal('netprinterStatus')}/>*/}
                                    <Col size={2}/>
                                </Row>
                                <Row size={3}/>
                            </Col>
                        </Row>
                        <Row size={0.75}>
                            <Col size={0.25}/>
                            <Col>
                                <Picker style={{color: '#fff'}}
                                        selectedValue={this.props.intl.locale}
                                        onValueChange={(itemValue, itemIndex) => {
                                            this.props.updateLang(itemValue);
                                            AsyncStorage.setItem('@PrimeDrive:last_lang', itemValue);
                                        }}>
                                    <Picker.Item label="English" value="en"/>
                                    <Picker.Item label="Danish" value="da"/>
                                </Picker>
                            </Col>
                            <Col size={0.25}/>
                        </Row>

                        <Row size={0.15}/>
                    </TouchableOpacity>
                </TouchableOpacity>

            </Modal>
        )
    }
}



