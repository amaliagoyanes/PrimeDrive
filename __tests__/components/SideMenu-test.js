'use strict';

import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';
import SideMenu from '../../src/components/SideMenu';


SideMenu.prototype.componentDidMount = jest.fn;


it('renders correctly', () => {
    const tree = shallow(
        <SideMenu/>);
    tree.setState({open: true});
    expect(tree).toMatchSnapshot();
    expect(tree.find('IconButton').length).toEqual(33);
    expect(tree.find('SwipeALot').length).toEqual(1);
    expect(tree.find('Icon[name="close"]').length).toEqual(1);
});
