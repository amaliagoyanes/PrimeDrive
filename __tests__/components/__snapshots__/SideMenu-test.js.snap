// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  onRequestClose={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.2)",
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  visible={true}
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.2)",
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        },
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={1}
      focusedOpacity={0.7}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#404048",
          "bottom": 0,
          "top": 0,
          "width": 250,
        }
      }
    >
      <RowNB
        size={1}
        style={
          Object {
            "alignSelf": "stretch",
            "height": 40,
            "marginTop": 30,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <ColumnNB
          size={2}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "marginTop": 5,
              }
            }
          >
            Menu
          </Text>
        </ColumnNB>
        <ColumnNB
          size={10}
        />
        <ColumnNB
          size={1}
        >
          <TouchableOpacity
            activeOpacity={0.2}
            focusedOpacity={0.7}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#3f3f48",
                  "height": 24,
                  "justifyContent": "center",
                  "width": 24,
                },
                Object {
                  "backgroundColor": "#53535a",
                  "marginTop": 0,
                },
              ]
            }
          >
            <Icon
              allowFontScaling={false}
              color="white"
              name="close"
              size={14}
            />
          </TouchableOpacity>
        </ColumnNB>
      </RowNB>
      <RowNB
        size={7}
      >
        <SwipeALot
          circleActiveStyle={
            Object {
              "backgroundColor": "#beb3b7",
            }
          }
          circleDefaultStyle={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#beb3b7",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 8,
              "margin": 7,
              "marginTop": 3,
              "top": -2,
              "width": 8,
            }
          }
        >
          <ColumnNB
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "marginBottom": 15,
                },
                Object {
                  "marginTop": 0,
                },
              ]
            }
          >
            <RowNB>
              <IconButton
                icon="printer"
                onPress={[Function]}
                price="Terminal"
              />
              <IconButton
                icon="check-circle-outline"
                onPress={[Function]}
                price="Check in/out"
              />
              <IconButton
                icon="close-circle-outline"
                onPress={[Function]}
                price="Closed Receipts"
              />
            </RowNB>
            <RowNB>
              <IconButton
                icon="pause"
                onPress={[Function]}
                price="Parked Receipts"
              />
              <IconButton
                icon="cash-multiple"
                price="Deposition"
              />
              <IconButton
                icon="package-variant-closed"
                price="Stock control"
              />
            </RowNB>
            <RowNB>
              <IconButton
                icon="chart-line"
                price="Statistics"
              />
              <IconButton
                icon="calendar-clock"
                onPress={[Function]}
                price="Open day"
              />
              <IconButton
                icon="calendar-remove"
                onPress={[Function]}
                price="Close day"
              />
            </RowNB>
            <RowNB>
              <IconButton
                icon="web"
                price="Website"
              />
              <IconButton
                icon="food-fork-drink"
                onPress={[Function]}
                price="Table management"
              />
              <ColumnNB />
            </RowNB>
            <RowNB
              size={2}
            />
          </ColumnNB>
          <ColumnNB
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "marginBottom": 15,
                },
                Object {
                  "marginTop": 0,
                },
              ]
            }
          >
            <RowNB>
              <IconButton
                icon="printer"
                onPress={[Function]}
                price="Terminal"
              />
              <IconButton
                icon="check-circle-outline"
                onPress={[Function]}
                price="Check in/out"
              />
              <IconButton
                icon="close-circle-outline"
                onPress={[Function]}
                price="Closed Receipts"
              />
            </RowNB>
            <RowNB>
              <IconButton
                icon="pause"
                onPress={[Function]}
                price="Parked Receipts"
              />
              <IconButton
                icon="cash-multiple"
                price="Deposition"
              />
              <IconButton
                icon="package-variant-closed"
                price="Stock control"
              />
            </RowNB>
            <RowNB>
              <IconButton
                icon="chart-line"
                price="Statistics"
              />
              <IconButton
                icon="calendar-clock"
                onPress={[Function]}
                price="Open day"
              />
              <IconButton
                icon="calendar-remove"
                onPress={[Function]}
                price="Close day"
              />
            </RowNB>
            <RowNB>
              <IconButton
                icon="web"
                price="Website"
              />
              <IconButton
                icon="food-fork-drink"
                onPress={[Function]}
                price="Table management"
              />
              <ColumnNB />
            </RowNB>
            <RowNB
              size={2}
            />
          </ColumnNB>
          <ColumnNB
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "marginBottom": 15,
                },
                Object {
                  "marginTop": 0,
                },
              ]
            }
          >
            <RowNB>
              <IconButton
                icon="printer"
                onPress={[Function]}
                price="Terminal"
              />
              <IconButton
                icon="check-circle-outline"
                onPress={[Function]}
                price="Check in/out"
              />
              <IconButton
                icon="close-circle-outline"
                onPress={[Function]}
                price="Closed Receipts"
              />
            </RowNB>
            <RowNB>
              <IconButton
                icon="pause"
                onPress={[Function]}
                price="Parked Receipts"
              />
              <IconButton
                icon="cash-multiple"
                price="Deposition"
              />
              <IconButton
                icon="package-variant-closed"
                price="Stock control"
              />
            </RowNB>
            <RowNB>
              <IconButton
                icon="chart-line"
                price="Statistics"
              />
              <IconButton
                icon="calendar-clock"
                onPress={[Function]}
                price="Open day"
              />
              <IconButton
                icon="calendar-remove"
                onPress={[Function]}
                price="Close day"
              />
            </RowNB>
            <RowNB>
              <IconButton
                icon="web"
                price="Website"
              />
              <IconButton
                icon="food-fork-drink"
                onPress={[Function]}
                price="Table management"
              />
              <ColumnNB />
            </RowNB>
            <RowNB
              size={2}
            />
          </ColumnNB>
        </SwipeALot>
      </RowNB>
      <RowNB
        size={0.5}
      />
    </TouchableOpacity>
  </TouchableOpacity>
</Modal>
`;
